### PR TITLE
Feature/bp 255/rebase on 2.4.0

### PR DESCRIPTION
**Pull request by engdorm**
flow-init feature

### CODE REVIEW

**Review by Mograbi — CHANGES_REQUESTED** [review]

File: setup.py [hunk 1]
```diff
@@ -4,9 +4,13 @@
...
 
 requirements = [
+    "async-timeout==3.0.1",
+    "aiohttp==3.6.2",
+    "aiohttp-cors==0.7.0",
     "aioredis==1.3.0",
     "uvloop==0.14.0",
-    "gd_node==2.4.*"
+    "gd_node==2.4.*",
+    "docker==5.0.3"
```
Comment (AlexFernandes-MOVAI) [inline setup.py:13]: We should use at least docker==6.0.0

**Review by AlexFernandes-MOVAI — CHANGES_REQUESTED** [review]

File: flow_initiator/spawner/container_tools/docker_base_manager.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+A file for docker base manager
+"""
+
+import socket
+from functools import wraps
+from typing import Callable, Optional
+
+from docker.errors import NotFound, APIError
+from flow_initiator.spawner.exceptions import OrchestratorException
+from movai_core_shared.envvars import DOCKER_REGISTRY
```
[the remark this answers is not in the document]
Comment (engdorm) [reply flow_initiator/spawner/container_tools/docker_base_manager.py:11]: there is a branch I need to merge

File: flow_initiator/spawner/async_spawner.py [hunk 3]
```diff
@@ -403,28 +410,28 @@ async def process_start(self, **kwargs):
...
             for lc_node in self.flow_monitor.load_ros2_lifecycle():
-                if lc_node[0] not in [command[0] for command in commmands_to_launch]:
-                    commmands_to_launch.append(lc_node)
+                if lc_node["node"] not in [command["node"] for command in commands_to_launch]:
+                    commands_to_launch.append(lc_node)
                 else:
-                    ros2_lifecycle_nodes_to_activate.append(lc_node[0])
-
-            # self._logger.info(commmands_to_launch)
+                    ros2_lifecycle_nodes_to_activate.append(lc_node["node"])
```
[the remark this answers is not in the document]
Comment (engdorm) [reply flow_initiator/spawner/async_spawner.py:414]: maybe 
it old code that we need to check

File: flow_initiator/spawner/flowmonitor.py [hunk 4]
```diff
@@ -122,8 +121,8 @@ def transition(
                 nodes_to_transit, self.active_flow, transition_msg=transition_msg
             )
 
-        except Exception as e:
-            LOGGER.error(e)
+        except Exception:
+            LOGGER.error()
```
Comment (engdorm) [inline flow_initiator/spawner/flowmonitor.py:125]: need to add e

Comment (engdorm) [reply flow_initiator/spawner/flowmonitor.py:125]: need to check why did I removed it

**Pull request closed without merging**